novice-r-practice.Rmd

```diff
@@ -0,0 +1,363 @@
+
+7. Use `dplyr::filter` to create a new table with a subset of rows by values:
+
+```{r filter-with-one-condition}
+dplyr::filter(person, family_name < "M")
+```
+
+*Create a table with only sites south of -48 degrees.*
+
+8. Use the pipe operator `%>%` to combine operations:
+
+```{r select-pipe-filter}
+person %>%
+  dplyr::select(family_name, personal_name) %>%
+  dplyr::filter(family_name < "M")
+```
+
+*Create a table with only the latitudes and longitudes of sites south of -48 degrees.*
+
+9. Use `dplyr::mutate` to create a new column with calculated values and `stringr::str_length` to calculate string length:
+
+```{r mutate-name-length}
+person %>%
+  dplyr::mutate(name_length = stringr::str_length(family_name))
+```
+
+*Use the built-in function `round` to create a table with latitudes and longitudes rounded to integers.*
+
+10. Use `dplyr::arrange` to order rows and (optionally) `dplyr::desc` to impose descending order:
+
+```{r mutate-and-arrange}
+person %>%
+  dplyr::mutate(name_length = stringr::str_length(family_name)) %>%
+  dplyr::arrange(dplyr::desc(name_length))
+```
+
+*Create a table sorted by decreasing longitude (i.e., most negative longitude last).*
+
+## Working with grouped data
+
+1. Read `survey/measurements.csv` and look at the data with `View`:
+
+```{r read-and-view}
+measurements <- readr::read_csv(here::here("novice-r-practice", "measurements.csv"))
+View(measurements)
+```
+
+2. Find rows where `reading` is not NA and saved as `cleaned`:
+
+```{r remove-reading-na}
+cleaned <- measurements %>%
+  dplyr::filter(!is.na(reading))
+cleaned
+```
+
+*Rewrite the filter expression to select rows where the visitor and quantity are not NA either.*
+
+```{r hidden-cleaned, echo=FALSE}
+cleaned <- measurements %>%
+  dplyr::filter(!is.na(visitor) & !is.na(quantity) & !is.na(reading))
+```
+
+3. Group measurements by quantity measured and count the number of each (the column is named `n` automatically):
+
+```{r group-by-quantity}
+cleaned %>%
+  dplyr::group_by(quantity) %>%
+  dplyr::count()
+```
+
+*Group by person and quantity measured.*
+
+4. Find the minimum, average, and maximum for each quantity:
+
+```{r min-ave-max}
+cleaned %>%
+  dplyr::group_by(quantity) %>%
+  dplyr::summarize(low = min(reading), mid = mean(reading), high = max(reading))
+```
+
+*Look at the range for each combination of person and quantity.*
+
+5. Rescale salinity measurements that are greater than 1:
+
+```{r rescale-salinity}
+cleaned <- cleaned %>%
+  dplyr::mutate(reading = ifelse(quantity == 'sal' & reading > 1.0, reading/100, reading))
+cleaned
+```
+
+*Do the same calculation use `case_when`.*
+
+6. Read `visited.csv`, drop the NAs, and join with the cleaned-up table of readings:
+
+```{r join-two-tables}
+cleaned <- readr::read_csv(here::here("novice-r-practice", "visited.csv")) %>%
+  dplyr::filter(!is.na(visit_date)) %>%
+  dplyr::inner_join(cleaned, by = c("visit_id" = "visit_id"))
```

Comment:
How much coverage of joins do you think is appropriate?  I.e. would a learner use `inner_join()` here because it is the only one they know about? Or is choosing an appropriate join part of this exercise?

A very useful skill is being able to see which rows don't join, i.e `anti_join()`, particularly when joining on columns that are subject to human data entry error.

Reply:
Cool - can you please suggest an exercise appropriate to this level?

Reply:
See https://github.com/gvwilson/merely-useful.github.io/pull/2

Reply:
thanks - merged.